### PR TITLE
Toolbox Develop

### PR DESCRIPTION
**Pull request by vuln3r**

### CODE REVIEW

**Comment by mbuchetics** [comment]
* please add the supported version to `Package.swift`, currently I can not run the tests using an iOS simulator because `... is not available in iOS 11 or newer`, macOS also doesn't work because UIKit doesn't exist there. I think minimum iOS 12 should be fine.
* don't use underscores in function names, even with tests it looks weird (`test_ArrayRemove_isRemoved_true` -> `testArrayRemove`)
* please add a README with a short description and info how to use it.

**Comment by mbuchetics** [comment]
* Formatters: number formatters are defined in a sub-struct, date formatters are not. this should be consistent. I would choose `Formatters.decimalNumber` over `Formatters.Numbers.decimal` but it's up to you.

File: Sources/Toolbox/Exstentions/Style.swift [hunk 1]
```diff
@@ -1,6 +1,7 @@
 
 import UIKit
 
+@available(iOS 11.0, *)
```
Comment (mbuchetics) [inline Sources/Toolbox/Exstentions/Style.swift:4]: let's just use iOS 12 as min, then you dont need the checks here.

File: Sources/Toolbox/Exstentions/SwiftUI/View+HideKeyboard.swift [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import SwiftUI
+
+#if canImport(UIKit)
```
Comment (mbuchetics) [inline Sources/Toolbox/Exstentions/SwiftUI/View+HideKeyboard.swift:3]: let's use iOS only, then you dont need the UIKit check

File: Sources/Toolbox/Utilities/SwiftUI/ImagePicker.swift [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+            parent.presentationMode.wrappedValue.dismiss()
+        }
+    }
+
+    @Binding var image: UIImage?
+    @Binding var sourceType: UIImagePickerController.SourceType
+    @SwiftUI .Environment(\.presentationMode) var presentationMode
+
+    let picker = UIImagePickerController()
```
Comment (mbuchetics) [inline Sources/Toolbox/Utilities/SwiftUI/ImagePicker.swift:28]: should use PHPickerViewController for iOS 14

File: Sources/Toolbox/Utilities/SwiftUI/ImagePicker.swift [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+    func alertPromptToAllowCameraAccessViaSetting() {
```
Comment (mbuchetics) [inline Sources/Toolbox/Utilities/SwiftUI/ImagePicker.swift:30]: the alert should be provided by the app, otherwise it's not localizable

**Review by mbuchetics — CHANGES_REQUESTED** [review]

**Comment by vuln3r** [comment]
> * Formatters: number formatters are defined in a sub-struct, date formatters are not. this should be consistent. I would choose `Formatters.decimalNumber` over `Formatters.Numbers.decimal` but it's up to you.

I ti is so in the old starter project I just copied it... but let's stick to one

File: Sources/Toolbox/Utilities/SwiftUI/ImagePicker.swift (hunk 3 above)
Comment (vuln3r) [reply Sources/Toolbox/Utilities/SwiftUI/ImagePicker.swift:28]: yes, I need to find/write a good example of it for `UIKit` and `SwiftUI` but now that ist for `iOS13` and `SwiftUI 1.0`

**Review by mbuchetics — APPROVED** [review]

**Comment by mbuchetics** [comment]
👍 I tagged the initial version as `1.0.0`

**Pull request merged**